src/main/groovy/com/twcable/grabbit/client/batch/steps/jcrnodes/JcrNodesWriter.groovy

```diff
@@ -88,8 +88,9 @@ class JcrNodesWriter implements ItemWriter<ProtoNode>, ItemWriteListener {
     }
 
     private static void writeToJcr(ProtoNode nodeProto, Session session) {
-        JcrNodeDecorator jcrNode = new ProtoNodeDecorator(nodeProto).writeToJcr(session)
-        jcrNode.setLastModified()
+        ProtoNodeDecorator protoNodeDecorator = new ProtoNodeDecorator(nodeProto)
+        JcrNodeDecorator nodeDecorator = new JcrNodeDecorator(protoNodeDecorator.getOrCreateNode(session))
+        nodeDecorator.writeToJcr(protoNodeDecorator, session)
         // This will processed all mandatory child nodes only
         if(nodeProto.mandatoryChildNodeList && nodeProto.mandatoryChildNodeList.size() > 0) {
```

Comment:
We are trying to keep application logic consolidated within our application objects (ProtoNodeDecorator, JcrNodeDecorator), and out of the framework (JcrNodesWriter). One thing can change without the other needing to know. In addition, we can use these objects in multiple places within the framework without worrying about the details.